# Vergleich zwischen realem Portfolio und Markowitz-Optimierung

**Autor:** Jean Jacques Kamdem  
**Matrikelnummer:** 548153  

---

## 1. Einführung

In der Praxis treffen Investoren ihre Entscheidungen häufig auf Basis von Erfahrung, Marktbeobachtungen und institutionellen Sicherheiten.

Ein Beispiel dafür sind Staatsanleihen in der CEMAC-Region, bei denen oft von einem sehr geringen Risiko ausgegangen wird.

In der Finanztheorie hingegen wird angenommen, dass jedes Investment mit Risiko verbunden ist und dass ein optimaler Kompromiss zwischen Risiko und Rendite existiert.

Ein bekanntes Modell hierfür ist die Markowitz-Portfolio-Optimierung.

Ziel dieses Notebooks ist es, ein praxisbasiertes Portfolio mit einem mathematisch optimierten Portfolio zu vergleichen.

## 2. Reales Portfolio (praxisbasiert)

Basierend auf Marktbeobachtungen nehmen wir folgende typische Renditen an:

- Staatsanleihen: 5.8%  
- Unternehmensanleihen: 6%  
- Aktien: 4%  

Ein Investor könnte intuitiv folgendes Portfolio wählen:

- 50% Staatsanleihen  
- 30% Unternehmensanleihen  
- 20% Aktien  

Dieses Portfolio basiert auf Erfahrung und Vertrauen, nicht auf mathematischer Optimierung.

In [32]:
import numpy as np

# Renditen
returns = np.array([0.058, 0.06, 0.04])

# Gewichte (reales Portfolio)
weights_real = np.array([0.5, 0.3, 0.2])

# Erwartete Rendite
expected_return_real = np.dot(weights_real, returns)

expected_return_real

np.float64(0.05500000000000001)

### Erklärung

- `returns` enthält die erwarteten Renditen der einzelnen Assets  
- `weights_real` beschreibt die Portfolio-Allokation  
- `np.dot()` berechnet das gewichtete Mittel (Portfolio-Rendite)  

Die erwartete Rendite des Portfolios wird als gewichtetes Mittel berechnet:

R = w₁r₁ + w₂r₂ + w₃r₃

Dabei sind:
- w die Gewichte
- r die Renditen der einzelnen Assets

3. Auswahl der Assets und Datenbasis

Im vorherigen Notebook wurde ein diversifiziertes Portfolio mit fünf Assets (SPY, QQQ, TLT, GLD, VNQ) verwendet, um die Markowitz-Optimierung unter realistischen Marktbedingungen zu analysieren.

Für den Vergleich mit einem praxisbasierten Portfolio wurde jedoch bewusst eine reduzierte Struktur mit drei zentralen Anlageklassen gewählt:
	•	Staatsanleihen (AGG)
	•	Unternehmensanleihen (LQD)
	•	Aktien (SPY)

Diese Vereinfachung ermöglicht einen besseren Vergleich mit realen Investitionsentscheidungen, da praxisbasierte Portfolios häufig aus wenigen Hauptanlageklassen bestehen.

Die Analyse basiert somit auf zwei komplementären Ebenen:
	•	Einer erweiterten Modellstruktur mit fünf Assets zur theoretischen Untersuchung der Markowitz-Optimierung
	•	Einer reduzierten Struktur mit drei Assets zur praxisnahen Analyse und besseren Interpretierbarkeit der Ergebnisse

Diese Reduktion stellt keinen Verlust an Modellqualität dar, sondern dient der klareren Abbildung realer Entscheidungsprozesse von Investoren.

Um ein realistisches Modell zu gewährleisten, werden historische Finanzdaten verwendet.
Da direkte Marktdaten aus der CEMAC-Region nur eingeschränkt verfügbar sind, werden börsengehandelte Fonds (ETFs) als Approximation der jeweiligen Anlageklassen genutzt.

Auf dieser Basis können Renditen und Risiken empirisch berechnet und anschließend im Rahmen der Portfoliooptimierung analysiert werden.


In [31]:
import yfinance as yf
import pandas as pd

assets = ["AGG", "LQD", "SPY"]

data = yf.download(assets, start="2020-01-01", end="2025-01-01")

# Auswahl der Schlusskurse
data = data["Close"]

data.head()

[*********************100%***********************]  3 of 3 completed


Ticker,AGG,LQD,SPY
Date,,,
2020-01-02,94.454132,103.002678,296.888184
2020-01-03,94.730759,103.356102,294.640137
2020-01-06,94.655319,103.010696,295.764160
2020-01-07,94.554733,102.745636,294.932526
2020-01-08,94.445755,102.601051,296.504364


### 4. Berechnung der Renditen

Auf Basis der historischen Schlusskurse berechnen wir die täglichen Renditen der Assets.
Diese dienen als Grundlage für die Risiko- und Portfolioanalyse.

In [9]:
# Tägliche Renditen berechnen
returns = data.pct_change().dropna()

returns.head()

Ticker,AGG,LQD,SPY
Date,,,
2020-01-03,0.002929,0.003431,-0.007572
2020-01-06,-0.000796,-0.003342,0.003815
2020-01-07,-0.001063,-0.002573,-0.002812
2020-01-08,-0.001153,-0.001407,0.005329
2020-01-09,0.001065,0.004776,0.006781


### 5. Erwartete Renditen und Kovarianzmatrix

Im nächsten Schritt berechnen wir die durchschnittlichen Renditen sowie die Kovarianzmatrix der Assets.
Diese Größen sind zentrale Bestandteile der Portfoliooptimierung nach Markowitz.

In [15]:
# Renditen (auf Basis realer Daten)
returns = data.pct_change().dropna()

# Durchschnittliche Renditen (annualisiert)
mean_returns = returns.mean() * 252

# Kovarianzmatrix (annualisiert)
cov_matrix = returns.cov() * 252

mean_returns, cov_matrix

(Ticker
 AGG   -0.001665
 LQD    0.003070
 SPY    0.155639
 dtype: float64,
 Ticker       AGG       LQD       SPY
 Ticker                              
 AGG     0.004665  0.006386  0.002653
 LQD     0.006386  0.012462  0.008237
 SPY     0.002653  0.008237  0.044089)

### 6. Reales Portfolio (empirische Bewertung)

Im Folgenden bewerten wir ein praxisbasiertes Portfolio anhand der zuvor berechneten Renditen und Risiken.

Die Gewichtung basiert auf einer typischen Allokation:
- 50% Staatsanleihen
- 30% Unternehmensanleihen
- 20% Aktien

In [17]:
import numpy as np

# Gewichte des realen Portfolios
weights_real = np.array([0.5, 0.3, 0.2])

# Erwartete Rendite
return_real = np.dot(weights_real, mean_returns)

# Risiko (Standardabweichung)
risk_real = np.sqrt(np.dot(weights_real.T, np.dot(cov_matrix, weights_real)))

return_real, risk_real

(np.float64(0.03121616740826873), np.float64(0.08652260499202122))

### 7. Markowitz-Optimierung

Im folgenden bestimmen wir das optimale Portfolio nach der Markowitz-Theorie.

Ziel ist es, die Gewichte der Assets so zu wählen, dass das Risiko minimiert wird bei gegebener erwarteter Rendite.

In [19]:
import scipy.optimize as sco

In [20]:
# Portfolio-Performance
def portfolio_performance(weights, mean_returns, cov_matrix):
    returns = np.dot(weights, mean_returns)
    risk = np.sqrt(np.dot(weights.T, np.dot(cov_matrix, weights)))
    return returns, risk

In [21]:
def portfolio_volatility(weights, mean_returns, cov_matrix):
    return portfolio_performance(weights, mean_returns, cov_matrix)[1]

In [22]:
# Nebenbedingungen
num_assets = len(mean_returns)

constraints = (
    {'type': 'eq', 'fun': lambda x: np.sum(x) - 1}  # Summe der Gewichte = 1
)

bounds = tuple((0, 1) for _ in range(num_assets))

# Startwerte
initial_weights = num_assets * [1. / num_assets]

In [23]:
optimal = sco.minimize(
    portfolio_volatility,
    initial_weights,
    args=(mean_returns, cov_matrix),
    method='SLSQP',
    bounds=bounds,
    constraints=constraints
)

weights_optimal = optimal.x
weights_optimal

array([0.95388571, 0.        , 0.04611429])

### 8. Vergleich der Portfolios

Im Folgenden vergleichen wir das praxisbasierte Portfolio mit dem optimalen Portfolio nach Markowitz.

Ziel ist es, Unterschiede in Rendite und Risiko zu analysieren.

In [24]:
# Performance des optimalen Portfolios
return_opt, risk_opt = portfolio_performance(weights_optimal, mean_returns, cov_matrix)

return_opt, risk_opt

(np.float64(0.005588538654713551), np.float64(0.06761420954756614))

In [25]:
print("Reales Portfolio:")
print("Rendite:", return_real)
print("Risiko:", risk_real)

print("\nOptimales Portfolio (Markowitz):")
print("Rendite:", return_opt)
print("Risiko:", risk_opt)

Reales Portfolio:
Rendite: 0.03121616740826873
Risiko: 0.08652260499202122

Optimales Portfolio (Markowitz):
Rendite: 0.005588538654713551
Risiko: 0.06761420954756614


### Interpretation der Ergebnisse

Das reale Portfolio erzielt eine höhere Rendite (ca. 3.12%), weist jedoch auch ein höheres Risiko auf (ca. 8.65%).

Im Gegensatz dazu minimiert das Markowitz-optimierte Portfolio das Risiko (ca. 6.76%), führt jedoch zu einer deutlich geringeren Rendite (ca. 0.56%).

Dies zeigt, dass eine rein mathematische Optimierung nicht zwangsläufig zu besseren praktischen Ergebnissen führt.

In der Praxis sind Investoren häufig bereit, ein höheres Risiko einzugehen, um bessere Renditen zu erzielen.

## 9. Sharpe Ratio Analyse

Um die Effizienz der Portfolios zu bewerten, verwenden wir den Sharpe Ratio.

Der Sharpe Ratio misst die Rendite im Verhältnis zum Risiko.

Formel:

Sharpe Ratio = (Rendite - risikofreier Zinssatz) / Risiko

Ein höherer Sharpe Ratio bedeutet eine bessere risikobereinigte Performance.

In [27]:
# Risikofreier Zinssatz (angenommen 1%)
risk_free_rate = 0.01

# Sharpe Ratio reales Portfolio
sharpe_real = (return_real - risk_free_rate) / risk_real

# Sharpe Ratio optimales Portfolio
sharpe_opt = (return_opt - risk_free_rate) / risk_opt

sharpe_real, sharpe_opt

(np.float64(0.24520953119968136), np.float64(-0.06524458948504036))

In [28]:
print("Sharpe Ratio (Reales Portfolio):", sharpe_real)
print("Sharpe Ratio (Optimales Portfolio):", sharpe_opt)

Sharpe Ratio (Reales Portfolio): 0.24520953119968136
Sharpe Ratio (Optimales Portfolio): -0.06524458948504036


### Interpretation

Das reale Portfolio weist einen höheren Sharpe Ratio auf als das Markowitz-optimierte Portfolio.

Dies bedeutet, dass das reale Portfolio eine bessere Rendite pro Risikoeinheit bietet.

Obwohl das optimale Portfolio ein geringeres Risiko hat, ist seine Rendite zu niedrig, um eine effiziente risikobereinigte Performance zu erzielen.

Dies zeigt die Grenzen der reinen Varianzminimierung in der Praxis.

## 10. Fazit

In dieser Arbeit haben wir ein praxisbasiertes Portfolio mit einem mathematisch optimierten Portfolio nach Markowitz verglichen.

Die Ergebnisse zeigen, dass das reale Portfolio eine höhere Rendite erzielt, jedoch auch mit einem höheren Risiko verbunden ist.

Das Markowitz-optimierte Portfolio hingegen minimiert das Risiko, führt aber zu einer deutlich geringeren Rendite.

Die Analyse des Sharpe Ratios hat gezeigt, dass das reale Portfolio eine bessere risikobereinigte Performance aufweist.

Dies deutet darauf hin, dass eine reine Minimierung der Varianz in der Praxis nicht immer optimal ist.

In realen Finanzmärkten spielen neben mathematischen Modellen auch Faktoren wie Marktbedingungen, Erwartungen und Risikobereitschaft eine wichtige Rolle.

Zusammenfassend lässt sich sagen, dass theoretische Modelle wie Markowitz wertvolle Orientierung bieten, jedoch nicht alle Aspekte realer Investitionsentscheidungen vollständig erfassen können.

# Für zukünftige Arbeiten könnte eine Erweiterung um Machine-Learning-Modelle interessante Ergebnisse liefern.